core/dbt/task/run.py

```diff
@@ -188,13 +190,17 @@ def after_execute(self, result):
 
     def _build_run_model_result(self, model, context):
         result = context['load_result']('main')
-        return RunModelResult(
+        adapter_query_status = {}
+        if isinstance(result.status, JsonSchemaMixin):
+            adapter_query_status = result.status.to_dict()
+        return RunResult(
             node=model,
             status=RunStatus.Success,
             timing=[],
             thread_id=threading.current_thread().name,
             execution_time=0,
-            message=result.status,
+            message=str(result.status),
+            adapter_query_status=adapter_query_status
```

Comment:
This effectively duplicates the `message` in `run_results.json`, nested within the result dict and also within the `adapter_query_status` dict. Are we doing this to avoid a breaking change for plugins / users that expect to print `message` as a key to `RunResult`?

Reply:
yes, that is correct. The `message` field on the run result could also contain error message so I think to be consistent it makes sense to keep this field. We could, however, remove the message field nested in the adapter response when we write out run results.

Reply:
> We could, however, remove the message field nested in the adapter response when we write out run results.

ok cool, I think this is the way to go, since `message` is more of a "dbt thing" and may or may not map to a specific adapter response value

Reply:
sounds good! I'll get this in and mark up for dev review